pkg/yang/entry.go

```diff
@@ -726,6 +726,30 @@ func (e *Entry) Path() string {
 	return e.Parent.Path() + "/" + e.Name
 }
 
+// Namespace returns the YANG/XML namespace Value for e as mounted in the Entry
+// tree (e.g., as placed by grouping statements).
+//
+// Per RFC6020 section 7.12, the namespace on elements in the tree due to a
+// "uses" statement is that of the where the uses statement occurs, i.e., the
+// user, rather than creator (grouping) of those elements, so we follow the
+// usage (Entry) tree up to the parent before obtaining the (then adjacent) root
+// node for its namespace Value.
+func (e *Entry) Namespace() *Value {
+	// Make e the root parent entry
+	for ; e.Parent != nil; e = e.Parent {
```

Comment:
Okay, this appears to be the correct logic.  In 7.12 it states:

>    The identifiers defined in the grouping are not bound to a namespace
>    until the contents of the grouping are added to the schema tree via a
>    "uses" statement that does not appear inside a "grouping" statement,
>    at which point they are bound to the namespace of the current module.

So a uses inside a grouping does not dictate the name space, only the uses in the main module do.  Submodules share the same name space as the main module, so even if they use a uses as the top level, it is still in the same name space as the main module.  Importing modules, on the other hand, only brings in definitions, it does not change the schema tree.  A uses inside of a group doesn't immediately change the schema tree, either.  The schema tree is only changed once the main module (or submodule) uses a uses outside a grouping.